arrow/arrow.py

```diff
@@ -29,6 +29,11 @@
     overload,
 )
 
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
```

Comment:
Shouldn't this be added to requirements for 3.11 and below?

https://github.com/arrow-py/arrow/tree/master/requirements

Reply:
A recent CR was merged to remove typing extensions for 3.8 and below. Mind rebasing this CR against the change and updating the requirements.txt to only pull it in for 3.9 and 3.10